phoenix-core/src/it/java/org/apache/phoenix/end2end/CursorWithRowValueConstructorIT.java

```diff
@@ -0,0 +1,687 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.end2end;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.phoenix.util.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.Properties;
+import java.util.Random;
+
+import static org.apache.phoenix.util.TestUtil.*;
+import static org.junit.Assert.*;
+
+
+public class CursorWithRowValueConstructorIT extends BaseClientManagedTimeIT {
+    private static final String TABLE_NAME = "CursorRVCTestTable";
+    protected static final Log LOG = LogFactory.getLog(CursorWithRowValueConstructorIT.class);
+
+    public void createAndInitializeTestTable() throws SQLException {
+        Connection conn = DriverManager.getConnection(getUrl());
+
+        PreparedStatement stmt = conn.prepareStatement("CREATE TABLE IF NOT EXISTS " + TABLE_NAME +
+                "(a_id INTEGER NOT NULL, " +
+                "a_data INTEGER, " +
+                "CONSTRAINT my_pk PRIMARY KEY (a_id))");
+        stmt.execute();
+        synchronized (conn){
+            conn.commit();
+        }
+
+        //Upsert test values into the test table
+        Random rand = new Random();
+        stmt = conn.prepareStatement("UPSERT INTO " + TABLE_NAME +
+                "(a_id, a_data) VALUES (?,?)");
+        int rowCount = 0;
+        while(rowCount < 100){
+            stmt.setInt(1, rowCount);
+            stmt.setInt(2, rand.nextInt(501));
+            stmt.execute();
+            ++rowCount;
+        }
+        synchronized (conn){
+            conn.commit();
+        }
+    }
+
+    public void deleteTestTable() throws SQLException {
+        Connection conn = DriverManager.getConnection(getUrl());
+        PreparedStatement stmt = conn.prepareStatement("DROP TABLE IF EXISTS " + TABLE_NAME);
+        stmt.execute();
+        synchronized (conn){
+            conn.commit();
+        }
+    }
+
+    @Test
+    public void testCursorsOnTestTablePK() throws SQLException {
+        try{
+            createAndInitializeTestTable();
+            String querySQL = "SELECT a_id FROM " + TABLE_NAME;
+
+            //Test actual cursor implementation
+            String cursorSQL = "DECLARE testCursor CURSOR FOR " + querySQL;
+            DriverManager.getConnection(getUrl()).prepareStatement(cursorSQL).execute();
+            cursorSQL = "OPEN testCursor";
+            DriverManager.getConnection(getUrl()).prepareStatement(cursorSQL).execute();
+            cursorSQL = "FETCH NEXT FROM testCursor";
```

Comment:
Is FETCH PRIOR tested at all? If not supported, should we remove it from the grammar for now?